owncloudApp/src/main/java/com/owncloud/android/shares/db/OCShare.kt

```diff
@@ -0,0 +1,170 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author David González Verdugo
+ * Copyright (C) 2019 ownCloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.shares.db
+
+import android.content.ContentValues
+import android.database.Cursor
+import androidx.room.ColumnInfo
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import com.owncloud.android.db.ProviderMeta.ProviderTableMeta
+import com.owncloud.android.lib.resources.shares.RemoteShare
+
+/**
+ * Represents one record of the Shares table.
+ */
+@Entity(tableName = ProviderTableMeta.OCSHARES_TABLE_NAME)
+data class OCShare(
+    @ColumnInfo(name = ProviderTableMeta.OCSHARES_FILE_SOURCE)
+    val fileSource: Long?,
```

Comment:
which params can be nullable? and which params will never be nullable?

Reply:
Fixed